src/compiler/crystal/compiler.cr

```diff
@@ -314,22 +314,29 @@ module Crystal
 
       target_machine.emit_obj_to_file llvm_mod, object_name
 
-      stdout.puts linker_command(program, object_name, output_filename, nil)
+      stdout.puts linker_command(program, [object_name], output_filename, nil)
     end
 
-    private def linker_command(program : Program, object_name, output_filename, output_dir)
+    private def linker_command(program : Program, object_names, output_filename, output_dir)
       if program.has_flag? "windows"
-        if object_name
-          object_name = %("#{object_name}")
-        else
-          object_name = %(%*)
-        end
-
         if link_flags = @link_flags.presence
           link_flags = "/link #{link_flags}"
         end
 
-        %(#{CL} #{object_name} "/Fe#{output_filename}" #{program.lib_flags} #{link_flags})
+        args = %(#{object_names.join(" ")} "/Fe#{output_filename}" #{program.lib_flags} #{link_flags})
+        cmd = "#{CL} #{args}"
+
+        if cmd.to_utf16.size > 32000
+          # The command line would be too big, pass the args through a UTF-16-encoded file instead.
+          # TODO: Do this the proper way when iconv is supported.
```

Comment:
Maybe a minor thing, but this comment insinuates that we can actually get `iconv` to work on win32 and be confident to use it in the compiler 🤷 Even then, this implementation might actually be "the proper way" we should be doing this because it doesn't need a library.
So I would suggest to put this code already in a dedicated method on `String`.

Reply:
Removed mention of "iconv".
Probably too early to think of such a method. Likely to be the only use case, and the proper API for it already exists separately, so I probably disagree.